sdk/python/tests/test_read.py

```diff
@@ -1,37 +1,148 @@
 """ Test read functions """
 
-from pvsite_datamodel.read.site import get_site, get_all_site, get_site_from_uuid
-from pvsite_datamodel.read.generation import get_pv_generation
+import uuid
+
+from sqlalchemy.orm import Query
+from typing import List
+
+from pvsite_datamodel.sqlmodels import SiteSQL, StatusSQL
+from pvsite_datamodel.read.site import get_all_sites
+from pvsite_datamodel.read.site import get_site_by_uuid
+from pvsite_datamodel.read.site import get_site_by_client_site_id
+from pvsite_datamodel.read.generation import get_pv_generation_by_sites
+from pvsite_datamodel.read.status import get_latest_status
+from pvsite_datamodel.read.generation import get_pv_generation_by_client
+from pvsite_datamodel.read.latest_forecast_values import get_latest_forecast_values_by_site
+
 import pytest
 
 
-def test_get_site(sites, db_session):
+class TestGetAllSites:
+    """Tests for the get_all_sites function"""
+
+    def test_returns_all_sites(self, sites, db_session):
+        out = get_all_sites(db_session)
+
+        assert len(out) == len(sites)
+        assert out[0] == sites[0]
+
+
+class TestGetSiteByUUID:
+    """Tests for the get_site_by_uuid function"""
+
+    def tests_gets_site_for_existing_uuid(self, sites, db_session):
+        site = get_site_by_uuid(session=db_session, site_uuid=sites[0].site_uuid)
+
+        assert site == sites[0]
+
+    def test_raises_error_for_nonexistant_site(self, sites, db_session):
+        with pytest.raises(Exception):
+            _ = get_site_by_uuid(session=db_session, site_uuid=uuid.uuid4())
+
+
+class TestGetSiteByClientSiteID:
+    """Tests for the get_site_by_client_site_id function"""
+
+    def test_gets_site_successfully(self, sites, db_session):
+        site = get_site_by_client_site_id(
+            session=db_session,
+            client_name="testclient_1",
+            client_site_id=1
+        )
+
+        assert site.client_site_id == 1
+
+    def test_raises_exception_when_no_such_site_exists(self, sites, db_session):
+        with pytest.raises(Exception):
+            _ = get_site_by_client_site_id(
+                session=db_session,
+                client_name="testclient_100",
+                client_site_id=1
+            )
+
+
+class TestGetPVGenerationByClient:
+    """Tests for the get_pv_generation_by_client function"""
+
+    def test_returns_all_generations_without_input_client(self, generations, db_session):
+        generations = get_pv_generation_by_client(session=db_session)
+        assert len(generations) == 40
+
+    def test_returns_all_generations_for_input_client(self, generations, db_session):
+        # TODO
```

Comment:
should we add this? or add a github task to do it?

Reply:
I can add it, it was a TODO from your previous PR and I just reformatted it into test skeletons

Reply:
Feel free to add a github issue and just reference it as a comment, then this PR wont be blocked

Reply:
[`7ff5a27` (#25)](https://github.com/openclimatefix/pvsite-datamodel/pull/25/commits/7ff5a2797532aadd3a99bee1eba54897ffa1fc79)